doc/source/overview/config_files.md

```diff
@@ -0,0 +1,753 @@
+In addition, many detectors also require (or allow) additional artefacts, such as kernels, functions and models. 
+Depending on their type, artefacts can be specified in `config.toml` in a number of ways: 
+
+- **Local files**: Simple functions and/or models can be specified as locally stored 
+[dill](https://dill.readthedocs.io/en/latest/dill.html) files, whilst data arrays are specified as locally stored
+numpy [npy](https://numpy.org/devdocs/reference/generated/numpy.lib.format.html) files.
+
+- **Function/object registry**: As discussed in [Registering artefacts](registering_artefacts), functions and other 
+objects defined at runtime can be registered using `alibi_detect.utils.registry`, allowing them to be specified 
+in the config file without having to serialise them. For convenience a number of Alibi Detect functions such as 
+[preprocess_drift](../api/alibi_detect.cd.tensorflow.preprocess.rst) are also pre-registered. 
+
+- **Dictionaries**: More complex artefacts are specified via nested dictionaries with a `src` field and 
+additional option/setting fields, which are sometimes further nested dictionaries themselves. See 
+[Artefact dictionaries](dictionaries) for further details.
+
+The following table shows the allowable formats for all artefacts that can be specified in a config file.
+
+```{table} Possible artefact formats
+:name: all-artefacts-table
+
+|Field                     |.npy file  |.dill file  |Registry|Dictionary| 
+|:-------------------------|:---------:|:----------:|:------:|:--------:|
+|`x_ref`                   |✔          |            |        |          |
+|`optimizer`               |           |✔           |✔       |          |
+|`reg_loss_fn`             |           |✔           |✔       |          |
+|`model`                   |           |            |✔       |✔         |
+|`preprocess_fn`           |           |✔           |✔       |✔         |
+|`preprocess_fn.model`     |           |            |✔       |✔         |
+|`preprocess_fn.embedding` |           |            |✔       |✔         |
+|`preprocess_fn.tokenizer` |           |            |✔       |✔         |
+|`dataset`                 |           |✔           |✔       |✔         |
+|`kernel`                  |           |✔           |✔       |✔         |
+|`kernel.proj`             |           |            |✔       |✔         |
+|`kernel.kernel_a`         |           |✔           |✔       |✔         |
+|`kernel.kernel_b`         |           |✔           |✔       |✔         |
+|`initial_diffs`           |✔          |            |        |          |
+```
+
+```{note}
+When TensorFlow, PyTorch, or scikit-learn `model`'s are specified, the model type should be specified via the `backend` field in the *config.toml*.
+For example, if `model` is a TensorFlow model, set `backend='tensorflow'. This is the case even for detectors that don't take a `backend` kwarg.
+```
```

Comment:
This doesn't render properly: 
![image](https://user-images.githubusercontent.com/13080878/160413211-b863f15a-273f-4662-8401-ea7b299f0ab3.png)

Also can you clarify the last sentence? Does it mean `backend='tensorflow'` should be put for everything by default?

Reply:
Ok, let's clarify on the public docs and perhaps leave a note somewhere in the source code about this behaviour?

Reply:
p.s. on a side note, this is related to the first "Outstanding Decisions" item. i.e. do we want to factor out `backend` into top-level detector one, `model` one, and `preprocess_fn` one. I don't think we need to right now, as mixing backends doesn't currently make sense. But it will become necessary when we introduce backends which are distinct from model types (e.g. `keops`). Things will also get more complicated here though, as for e.g. `keops` supports PyTorch but not TensorFlow (so the detector `backend` will restrict the options for the `preprocess_fn` "backend"). 

Given the complexity, I'm tempted to leave for now (and just note as you mentioned)?